framewirc/messages.py

```diff
@@ -79,37 +79,35 @@ def _chunk_message(message, max_length):
     lines = deque(message.splitlines())
     while lines:
         line = lines.popleft()
-        line_bytes = to_bytes(line)
+        line_bytes = line.encode()
         # If the line fits, add it the the lines.
         if len(line_bytes) < max_length:
             yield line_bytes
             continue
 
-        # Whole line doesn't fit, so see if it can be split on space.
-        letterpoint = None  # Where we should break if there is no space
-        spacepoint = None  # Where we should break if there is a space.
+        # See if there is a space below the max length.
+        spacepoint = line_bytes.rfind(b' ', 0, max_length+1)
+        if spacepoint != -1:
+            # Break on the last space that fits.
+            start = line_bytes[:spacepoint]
+            yield start
+            # ... and add what's left back into the line pool.
+            end = line_bytes[(spacepoint + 1):].decode()
+            lines.appendleft(end)
+            # And move onto the next line.
+            continue
+
+        # Whole line does not contain spaces, so split within word.
         line_length = 0  # The running total size of the line
         for i, str_char in enumerate(line):
-            char_length = len(to_bytes(str_char))
+            char_length = len(str_char.encode())
             line_length += char_length
-            if str_char == ' ':
-                spacepoint = i
             if line_length > max_length:
                 letterpoint = i
                 break
```

Comment:
Am I correct that you're yielding utf-8 encoded bytes? If so, you can rely on the details of the utf-8 encoding to check only the last 4 bytes before your max length to find where the last fitting character starts. See https://en.wikipedia.org/wiki/UTF-8#Description

Something like this works:

```python
# Prefixes to utf-8 character codes
ONE_BYTE = 0
TWO_BYTES = 0b110
THREE_BYTES = 0b1110
FOUR_BYTES = 0b11110


def find_first_invalid_char(utf8_bytes):
    """
    Return the negative index of the first invalid character in utf8_bytes, or 0

    The bits of the first byte of each utf-8 encoded character encode how long the character is:
        One byte: 0xxxxxxx
        Two bytes: 110xxxxx
        Three bytes: 1110xxxx
        Four bytes: 11110xxx
        Continuation byte: 10xxxxxx

    If the last character started in `utf8_bytes` is completed we return 0, otherwise we return the
    (negative) index where the last character starts.
    """
    b1, b2, b3, b4 = utf8_bytes[-4:]
    # Last character doesn't cross max_length boundary
    if b4 >> 7 == ONE_BYTE or b3 >> 5 == TWO_BYTES or b2 >> 4 == THREE_BYTES or b1 >> 3 == FOUR_BYTES:
        return 0
    # b4 is invalid
    elif b4 >> 5 == TWO_BYTES or b4 >> 4 == THREE_BYTES or b4 >> 3 == FOUR_BYTES:
        return -1
    # b3 is invalid
    elif b3 >> 4 == THREE_BYTES or b3 >> 3 == FOUR_BYTES:
        return -2
    # b2 is invalid
    return -3


letterpoint = max_length + find_first_invalid_char(line_bytes)
```

You may want to rearrange this for efficiency - do all `TWO_BYTE` checks before `THREE_BYTE` checks, etc - but I think this is most understandable. I also wondered if there was a more efficient way to do the bit checking that avoids needing to do a different bitshift for each comparison, but didn't think of one.